service/storage_service.cc

```diff
@@ -3766,17 +3773,17 @@ storage_service::get_all_ranges(const std::vector<token>& sorted_tokens) const {
 }
 
 inet_address_vector_replica_set
-storage_service::get_natural_endpoints(const sstring& keyspace,
+storage_service::get_natural_endpoints_in_thread(const sstring& keyspace,
```

Comment:
It's outside the scope of the patch. I am thinking about if we should  get rid of the get_natural_endpoints_in_thread completely. It is used only once by api/storage_service.cc, considering we are in the direction of making storage_service slim.

Reply:
it makes sense.
Even the call to `partition_key::from_nodetool_style_string` makes much more sense in the api layer than here.

Reply:
Done in v2